imagepipeline-base/src/main/java/com/facebook/imageformat/ImageFormat.java

```diff
@@ -32,7 +35,8 @@
      * @return the image format or null if unknown
      */
     @Nullable
-    ImageFormat determineFormat(byte[] headerBytes, int headerSize);
+    ImageFormat determineFormat(byte[] headerBytes, int headerSize, InputStream is)
```

Comment:
This is quite a breaking API change and further down you're reading the whole input stream in a byte array. Can you not just read the first N bytes instead like we do for all other formats?